autorest/client.go

```diff
@@ -202,19 +203,21 @@ func (c Client) Do(r *http.Request) (*http.Response, error) {
 	if err != nil {
 		return nil, NewErrorWithError(err, "autorest/Client", "Do", nil, "Preparing request failed")
 	}
-	resp, err := SendWithSender(c.sender(), r)
+
+	if c.Sender == nil {
+		c.Sender = sender()
+	}
+
+	resp, err := SendWithSender(c.Sender, r)
 	Respond(resp,
 		c.ByInspecting())
 	return resp, err
 }
 
 // sender returns the Sender to which to send requests.
-func (c Client) sender() Sender {
-	if c.Sender == nil {
```

Comment:
I'm not convinced that removing this from here was a good choice as I now see this nil check sprinkled in various places.